src/lib/Libifl/pbsD_connect.c

```diff
@@ -357,63 +248,46 @@ __pbs_connect_extend(char *server, char *extend_data)
 			setenv("SystemRoot", "C:\\WINDOWS", 1);
 		}
 #endif
-		sock = socket(AF_INET, SOCK_STREAM, 0);
+	sd = socket(AF_INET, SOCK_STREAM, 0);
+	if (sd == -1) {
+		pbs_errno = PBSE_SYSTEM;
+		return -1;
+	}
 
+	pbs_strncpy(pbs_server, server, sizeof(pbs_server)-1); /* set for error messages from commands */
+	pbs_server[sizeof(pbs_server) - 1] = '\0';
 		/* and connect... */
 
-		if (have_alt) {
-			server = altservers[i];
-		}
-		strcpy(pbs_server, server); /* set for error messages from commands */
-
-		/* If a specific host name is defined which the client should use */
-
-		if (pbs_conf.pbs_public_host_name) {
-			/* my address will be in my_sockaddr,  bind the socket to it */
-			my_sockaddr.sin_port = 0;
-			if (bind(sock, (struct sockaddr *)&my_sockaddr, sizeof(my_sockaddr)) != 0) {
-				return -1;
-			}
-		}
-
-		if (get_hostsockaddr(server, &server_addr) != 0)
+	/* If a specific host name is defined which the client should use */
```

Comment:
What does this comment mean?

Reply:
@bremanandjk  can you please clarify?

Reply:
It seems the code associated with this comment is refactored or removed. Probably we can remove this line.